src/main/java/duke/bot/Ui.java

```diff
@@ -0,0 +1,120 @@
+     * */
+    public String constructGoodbyeMsg() {
+        return constructMsg("Meow. Hope to see you again soon!");
+    }
+
+    /**
+     * Constructs a message to indicate the addition of a new task
+     *
+     * @param task Task that was added
+     * @param tasksSize Number of total tasks after adding that new task
+     * @return A formatted string containing the constructed task addition message
+     */
+    public String constructAddMsg(Task task, int tasksSize) {
+        String msg = "Got it meow. I've added this task:\n" + String.format("  [%s][%s] %s\n", task.getTypeSymbol(),
+                task.getStatusSymbol(), task.getDesc()) + String.format("Now you have %d tasks in the list.\n",
+                tasksSize);
+        return constructMsg(msg);
+    }
+
+    /**
+     * Constructs a message to indicate the completion of a task
+     *
+     * @param index Index of the completed task in the list
+     * @param task Task that was completed
+     * @return A formatted string containing the constructed task completion message
+     */
+    public String constructDoneMsg(int index, Task task) {
+        String msg = "Good job meow, I've marked this task as done:\n" + String.format("%d.[%s][%s] %s\n", index + 1,
+                task.getTypeSymbol(), task.getStatusSymbol(), task.getDesc());
+        return constructMsg(msg);
+    }
+
+    /**
+     * Constructs a message to indicate the deletion of a task
+     *
+     * @param task Task that was deleted
+     * @param tasksSize Number of total tasks left after deleting that task
+     * @return A formatted string containing the constructed task deletion message
+     */
+    public String constructDeleteMsg(Task task, int tasksSize) {
+        String msg = "Noted meow. I've removed this task:\n" + String.format("  [%s][%s] %s\n", task.getTypeSymbol(),
+                task.getStatusSymbol(), task.getDesc()) + String.format("Now you have %d tasks in the list.\n",
+                tasksSize);
+        return constructMsg(msg);
+    }
+
+    /**
+     * Constructs a message containing all tasks found from the search
+     *
+     * @param tasks A list of tasks as search results
+     * @return A formatted string containing the constructed search result message
+     */
+    public String constructFoundMsg(List<Task> tasks) {
+        String msg = "Meow, here are the matching tasks in your list:\n";
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            msg += String.format("%d.[%s][%s] %s\n", i + 1, task.getTypeSymbol(), task.getStatusSymbol(),
+                    task.getDesc());
+        }
+        return constructMsg(msg);
+    }
+
+    /**
+     * Constructs a message containing all tasks found in the list
+     *
+     * @param tasks List of tasks
+     * @return A formatted string containing the constructed task list message
+     */
+    public String constructTaskList(List<Task> tasks) {
```

Comment:
This method doesn't serve to construct TaskList in my opinion, perhaps naming it as displayTaskList would sound more reasonable. What do you think